tests/test_permissions.py

```diff
@@ -0,0 +1,1254 @@
+    field_id('Mutation', 'approveShareExtension'): TestData(
+        tenant_perm=MANAGE_SHARES, resource_perm=APPROVE_SHARE_OBJECT
+    ),
+    field_id('Mutation', 'approveShareObject'): TestData(tenant_perm=MANAGE_SHARES, resource_perm=APPROVE_SHARE_OBJECT),
+    field_id('Mutation', 'approveTermAssociation'): TestData(
+        tenant_perm=MANAGE_GLOSSARIES, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'archiveOrganization'): TestData(
+        tenant_perm=MANAGE_ORGANIZATIONS, resource_perm=DELETE_ORGANIZATION
+    ),
+    field_id('Mutation', 'batchMetadataFormFieldUpdates'): TestData(
+        tenant_perm=MANAGE_METADATA_FORMS, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'cancelShareExtension'): TestData(
+        tenant_perm=MANAGE_SHARES, resource_perm=SUBMIT_SHARE_OBJECT
+    ),
+    field_id('Mutation', 'createAttachedMetadataForm'): TestData(
+        tenant_ignore=IgnoreReason.BACKPORT, resource_perm=ATTACH_METADATA_FORM
+    ),
+    field_id('Mutation', 'createCategory'): TestData(
+        tenant_perm=MANAGE_GLOSSARIES, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'createDataPipeline'): TestData(tenant_perm=MANAGE_PIPELINES, resource_perm=CREATE_PIPELINE),
+    field_id('Mutation', 'createDataPipelineEnvironment'): TestData(
+        tenant_perm=MANAGE_PIPELINES, resource_perm=CREATE_PIPELINE
+    ),
+    field_id('Mutation', 'createDataset'): TestData(tenant_perm=MANAGE_DATASETS, resource_perm=CREATE_DATASET),
+    field_id('Mutation', 'createDatasetStorageLocation'): TestData(
+        tenant_perm=MANAGE_DATASETS, resource_perm=CREATE_DATASET_FOLDER
+    ),
+    field_id('Mutation', 'createEnvironment'): TestData(
+        tenant_perm=MANAGE_ENVIRONMENTS, resource_perm=LINK_ENVIRONMENT
+    ),
+    field_id('Mutation', 'createGlossary'): TestData(
+        tenant_perm=MANAGE_GLOSSARIES, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'createMetadataForm'): TestData(
+        tenant_perm=MANAGE_METADATA_FORMS, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'createMetadataFormFields'): TestData(
+        tenant_perm=MANAGE_METADATA_FORMS, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'createMetadataFormVersion'): TestData(
+        tenant_perm=MANAGE_METADATA_FORMS, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'createNetwork'): TestData(tenant_perm=MANAGE_ENVIRONMENTS, resource_perm=CREATE_NETWORK),
+    field_id('Mutation', 'createOmicsRun'): TestData(tenant_perm=MANAGE_OMICS_RUNS, resource_perm=CREATE_OMICS_RUN),
+    field_id('Mutation', 'createOrganization'): TestData(
+        tenant_perm=MANAGE_ORGANIZATIONS, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'createQuicksightDataSourceSet'): TestData(
+        tenant_ignore=IgnoreReason.TENANT, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
```

Comment:
`'createQuicksightDataSourceSet'` is using `is_tenant_admin` in an indirect way. Can we add the `tenant_admin_perm` flag?